### PR TITLE
Allow for running tests in a host app

### PR DESCRIPTION
**Pull request by dfed**
In order to run tests in a host app in Xcode, the unit test target and the host app must be part of the same `xcodeproj`, and therefore they must be defined in the same `BUCK` file.

While I was here, I also added the ability to specify a `destination_specifier` on a test library.

Note that this still does not get UI tests working. But this would allow for us to have a unit test that required a host app

### CODE REVIEW

File: App/BUCK [hunk 1]
```diff
@@ -272,3 +270,39 @@ apple_resource(
...
+        ":ExampleAppLibrary",
+    ],
+)
+
+### Tests End ###
+
+### UI Tests Begin ###
+
+apple_test_lib(
+    name = "UITests",
```
Comment (qyang-nj) [inline App/BUCK]: Unit tests with a host application is actually not UI tests. UI tests cannot access the code of the app, but unit tests can. I'm not sure if Buck supports UI tests.

Comment (dfed) [reply App/BUCK]: I’m pretty sure Buck does not support UI tests per #18. But this was already here under the `App/UITests/` directory. We should probably remove this test (which is actually a UI test, and does fail), and instead replace it with something akin to our screenshot or accessibility tests.

Comment (thedavidharris) [reply App/BUCK]: @dfed Buck at least recognizes the extension https://github.com/facebook/buck/blob/a6c7d7391a7a8e575c6372cac6690a240b5907ea/test/com/facebook/buck/apple/testdata/apple_test_xcuitest/BUCK.fixture#L64, `is_ui_test` isn't documented though. Haven't tried it.

Comment (dfed) [reply App/BUCK]: TIL! Thanks for surfacing. We'll give it a shot.

Comment (fdiaz) [reply App/BUCK]: I did try using `is_ui_test` without success in our app, but hey, we could try again here?

Comment (dfed) [reply App/BUCK]: Shockingly, that worked: https://github.com/airbnb/BuckSample/pull/70. Thanks!!

**Pull request merged**